Core/Graphics/Source/FrameBuffer.cpp

```diff
@@ -53,37 +53,69 @@ namespace Babylon::Graphics
 
     void FrameBuffer::Clear(bgfx::Encoder& encoder, uint16_t flags, uint32_t rgba, float depth, uint8_t stencil)
     {
-        // BGFX requires us to create a new viewID, this will ensure that the view gets cleaned.
+        // BGFX requires us to create a new viewID, this will ensure that the view gets cleared.
         m_viewId = m_context.AcquireNewViewId(encoder);
 
         bgfx::setViewMode(m_viewId.value(), bgfx::ViewMode::Sequential);
         bgfx::setViewClear(m_viewId.value(), flags, rgba, depth, stencil);
         bgfx::setViewFrameBuffer(m_viewId.value(), m_handle);
 
-        // BGFX will consider the viewport when cleaning the screen, but WebGL always cleans the entire screen.
-        // That's why we always set the viewport to {0, 0, 1, 1} when cleaning.
-        bgfx::setViewRect(m_viewId.value(), 0, 0, Width(), Height());
+        // If a scissor is set, we need to set the BGFX view rect to match it before clearing.
+        // We set the view rect instead of the view scissor because BGFX clears after the view rect is set and before
+        // the view scissor is set.
+        // 
+        // If no scissor is set, WebGL cleans the entire screen, so set the view rect to cover the entire screen before
+        // clearing to match WebGL's behavior; otherwise BGFX will only clear the view rect.
+        //
+        // Note that the view rect and view scissor are reset to the desired dimensions before the encoder is submitted.
+        if (m_desiredScissor.Width != 0.0 || m_desiredScissor.Height != 0.0)
+        {
+            const auto flippedScissor = GetFlippedScissor();
+            bgfx::setViewRect(
+                m_viewId.value(),
+                static_cast<uint16_t>(flippedScissor.X),
+                static_cast<uint16_t>(flippedScissor.Y),
+                static_cast<uint16_t>(flippedScissor.Width),
+                static_cast<uint16_t>(flippedScissor.Height));
```

Comment:
This is currently truncated. Should it round?

Reply:
Hmm, maybe we should pass these from the JS side as ints instead of floats? WebGL uses ints for scissors but Babylon.js `Engine.enableScissor` uses floats.

Reply:
If we stay with floats then I'm not sure if we should round or not. I think since WebGL uses ints then it's ok to truncate.

Reply:
There is no such thing as pass as ints since JS doesn't have ints. JS only has doubles (well, a form of a double).

> I think since WebGL uses ints then it's ok to truncate.

I'm not sure I understand this statement. `GLint` is just an alias for `Number`. We need to find out what the underlying behavior is supposed to do. Maybe we can test it and see?

Reply:
I tried out some fractional values in Babylon.js and Spector shows they're being rounded, not truncated. Good call.